notebook/gateway/managers.py

```diff
@@ -586,6 +598,7 @@ def get_kernel_spec_resource(self, kernel_name, path):
             The name of the desired resource
         """
         kernel_spec_resource_url = url_path_join(self.base_resource_endpoint, str(kernel_name), str(path))
+        kernel_spec_resource_url = self._get_endpoint_for_user_filter(kernel_spec_resource_url)
```

Comment:
We're not applying filtering to the resource files (typically icons).  These are fetched only after the set of (filtered) kernelspecs have been returned, so are essentially filtered already.

Reply:
Ok, I have deleted it